Added keyboard shortcuts

eliaslfox opened the pull request:
I really like this, but I though keyboard shortcuts could make it better

Commit:
Added keyboard shortcuts

vishaltelangre commented on src/Main.elm:
```diff
@@ -376,6 +381,35 @@ view model =
+
+                ( 39, RemoteData.Success ( latestNumber, requestedNumber ) ) ->
+                    if requestedNumber /= latestNumber then
+                        LoadRequestedComic <| requestedNumber + 1
```
Similar logic exists in `viewNavigation` function to get the previous and next comic links. It would be great if you can extract it into smaller functions and reuse at both the places.

vishaltelangre replied:
Can we also show navigation buttons as follows just to help the visitors that it exists?

![xkcd comic browser 2017-12-31 18-08-39](https://user-images.githubusercontent.com/876195/34461652-c0820666-ee55-11e7-88e3-f7db20fcb9ef.png)

Here's the modified markup it needs:

```diff
 <div class="navigation">
-  <a href="#/comics/1014">Previous</a>
-  <a href="#">Latest</a>
-  <a href="#/comics/random">Random</a>
-  <a href="#/comics/1016">Next</a>
+  <a href="#/comics/1014" style="text-decoration: none;text-align: center;display: inline-block;vertical-align: middle;">
+    <span style="text-decoration: underline;">Prev</span>
+    <br>
+    <code style="font-family: monospace;border: 1px solid #000;border-radius: 5px;padding: 0 5px 2px;color: #fff;background: #000;">←</code>
+  </a>
+  <a href="#" style="text-decoration: none;text-align: center;display: inline-block;vertical-align: middle;">
+    <span style="text-decoration: underline;">Latest</span>
+    <br>
+    <code style="font-family: monospace;border: 1px solid #000;border-radius: 5px;padding: 0 5px 2px;color: #fff;background: #000;">H</code>
+  </a>
+  <a href="#/comics/random" style="text-decoration: none;text-align: center;display: inline-block;vertical-align: middle;">
+    <span style="text-decoration: underline;">Random</span>
+    <br>
+    <code style="font-family: monospace;border: 1px solid #000;border-radius: 5px;padding: 0 5px 2px;color: #fff;background: #000;">R</code>
+  </a>
+  <a href="#/comics/1016" style="text-decoration: none;text-align: center;display: inline-block;vertical-align: middle;">
+    <span style="text-decoration: underline;">Next</span>
+    <br>
+    <code style="font-family: monospace;border: 1px solid #000;border-radius: 5px;padding: 0 5px 2px;color: #fff;background: #000;">→</code>
+  </a>
 </div>
```

We will need to map `h/H` keypress to `Latest` and `r/R` keypress to `Random` link.

vishaltelangre commented on src/Main.elm:
```diff
@@ -376,6 +381,35 @@ view model =
+                        Noop
+
+                _ ->
+                    Noop
```
`Noop` value isn't really required, IMO.

eliaslfox replied:
How else would you handle the subscription when the key pressed shouldn't do anything?

vishaltelangre replied:
Oh yes. I totally forgot that. Never mind.

vishaltelangre reviewed (changes requested):
Thank you very much @eliaslfox. 

I really like this feature. 

Can you please take care of the suggested feedback. We will be good to merge it once that is done.

eliaslfox commented:
Okay, I'll add the changes in a bit

Commit:
cleaned up code

Commit:
more stuff I guess

Commit:
Merge branch 'master' of https://github.com/vishaltelangre/elm-xkcd-browser

eliaslfox commented:
I cleaned up the code a bit, and added the icons. It works but the arrows are different sizes. Should all of the icons have the same styling? 

Also `Main.viewNavigation` was taking in `0 0` in the event of an error, which relies on implementation details of the function to work properly. I changed it so now `viewNavigation` takes in a maybe type.

vishaltelangre commented:
Thanks a lot! Looks good, merging.

vishaltelangre commented:
It is deployed at https://vishaltelangre.com/xkcd and looks good. 

![image](https://user-images.githubusercontent.com/876195/34485764-409dcbf8-eff2-11e7-8bfc-c9aa469e1588.png)
